crypto/signer.go

```diff
@@ -31,39 +31,39 @@ var sigpbHashLookup = map[crypto.Hash]sigpb.DigitallySigned_HashAlgorithm{
 // Signer is responsible for signing log-related data and producing the appropriate
 // application specific signature objects.
 type Signer struct {
-	hash   crypto.Hash
```

Comment:
Should `NewSigner` move to the tree configuration object so it can pick the right hash?